src/server_manager/ui_components/outline-server-settings.html

```diff
@@ -208,6 +245,77 @@ <h3>[[localize('settings-metrics-header')]]</h3>
           this.fire(metricsSignal);
         }
       },
+      _setKeysPortButtonsVisible: function(visible) {
```

Comment:
When we create a UI component, we need to think about it's public interface: how do we interact with it.

There are 3 states for an editable input:
- Saved
- Editing (with or without error)
- Saving

There are a few operations:
- Edit (Saved -> Editing, on widget tap)
- Save (Editing -> Saving, on save tap, disables widget and fires event to app)
- CancelEdit (Editing -> Saved, on cancel tap)
- EditInvalidInput (Saving|Editing -> Editing, shows error, called by app)
- Commit (Saving|Editing -> Saved, updates view, called by app)

Let's make sure that's represented in code and create a public method for each of those operations. It will make it easier to follow the logic and allow us to move more code to the Typescript code. Furthermore, it allows for easier testing without having to have do magic with the DOM tree.

It may be worth mentioning the interface above as a comment.